### PR TITLE
Change TablesNeedUpgradeSysCheck to check for re-index requirement

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

The check with ID 3 was initially the sys-check that checked for tables
that need to be re-created (See commit da9a66e918b57c487d56a6ec3ae6bbed60b7db93)

This changes the logic of the check back to check for the re-index
requirement as a segment upgrade alone isn't sufficient to make tables
compatible for the next major version.

It also changes the severity to low (given that no action has to be
taken if there are no immediate upgrade plans).

Within CrateDB 4.x this will trigger for tables created in 3.x.
With a backport to 3.3 this will trigger for tables created in 2.x.

## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: blackbox/docs/appendices/release-notes/unreleased.rst [hunk 1]
```diff
@@ -46,6 +46,11 @@ None
 Fixes
 =====
 
+- The sys check that checks tables for compatibility with future versions now
```
Comment (andreidan) [inline blackbox/docs/appendices/release-notes/unreleased.rst]: This reads a bit difficult and I think the segments detail is not particularly useful. I'd rephrase this to something along the lines of "Fixed tables compatibility check to correctly indicate when tables need to be recreated in preparation for a CrateDB upgrade towards a major version" or something like that

**Comment by mfussenegger** [comment]
> but I'd drop the "re-index" and replace it with "recreate table"

right, I've adapted the wording.
